public/app/features/query/components/QueryEditorRow.tsx

```diff
@@ -243,7 +252,7 @@ export class QueryEditorRow<TQuery extends DataQuery> extends PureComponent<Prop
   isWaitingForDatasourceToLoad(): boolean {
     // if we not yet have loaded the datasource in state the
     // ds in props and the ds in state will have different values.
-    return this.props.dataSource.uid !== this.state.loadedDataSourceIdentifier;
+    return this.getInterpolatedDataSourceUID() !== this.state.loadedDataSourceIdentifier;
```

Comment:
This is solving the issue of not displaying the editor for certain scenarios #64011. 

But I found it confusing that `this.state.loadedDataSourceIdentifier` contains the value we're trying to load, even if the datasource doesn't exist. When it doesn't exist, the loaded data source will be the default one, but the value in this state property will be `this.getInterpolatedDataSourceUID()`.

This is very confusing, so maybe we can just change the name of the state property to `queriedDataSource`. WDYT?

PS: _I spent a considerable amount of time understanding why the state value is different than the one rendering the editor_

Reply:
@ivanortegaalba hm.. if you get the default data source when asking for a one that does not exist, maybe we should change that?

Reply:
Just to be clear, when we call `this.dataSourceSrv.get(interpolatedUID)` it fails. So the logic for dataSourceSrv is correct. 
We get the default datasource [here](https://github.com/grafana/grafana/pull/62617/files#diff-d941f1ab9516ae305170c05387d11f8cb39972e8a67e001c1840b7f6985710a1R156-R161) once the request has failed, we just assign the default one. But the `loadedDataSourceIdentifier` won't have the default datasource UID. This was the main confusion